src/components/message/message-card/MessageCard.tsx

```diff
@@ -39,157 +34,168 @@ export interface RecoveredProps {
-	const { sortOrderItems } = useMessageBodySortStore();
-
-	const [isHighlighted, setHighlighted] = React.useState(false);
-
-	const highlightTimer = React.useRef<NodeJS.Timeout>();
-	const hoverTimeout = React.useRef<NodeJS.Timeout>();
-
-	const isContentBeautified = messagesStore.beautifiedMessages.includes(messageId);
-	const isBookmarked =
-		selectedStore.bookmarkedMessages.findIndex(
-			bookmarkedMessage => bookmarkedMessage.id === messageId,
-		) !== -1;
-
-	const isSoftFiltered = messagesDataStore.isSoftFiltered.get(messageId);
-	const isDetailed = messagesStore.detailedRawMessagesIds.includes(messageId);
-
-	React.useEffect(() => {
-		const abortController = new AbortController();
-
-		if (
-			messagesStore.filterStore.isSoftFilter &&
-			messagesDataStore.isSoftFiltered.get(messageId) === undefined
-		) {
-			messagesDataStore.matchMessage(messageId, abortController.signal);
-		}
-
-		return () => {
-			abortController.abort();
-		};
-	}, []);
-
-	React.useEffect(() => {
-		switch (viewType) {
-			case MessageViewType.FORMATTED:
-				messagesStore.beautify(messageId);
-				break;
-			case MessageViewType.ASCII:
-				messagesStore.hideDetailedRawMessage(messageId);
-				break;
-			case MessageViewType.BINARY:
-				messagesStore.showDetailedRawMessage(messageId);
-				break;
-			default:
-				messagesStore.debeautify(messageId);
-				break;
-		}
-	}, [viewType]);
-
-	React.useEffect(() => {
-		if (!isHighlighted) {
-			if (messagesStore.highlightedMessageId === messageId) {
-				setHighlighted(true);
-
-				highlightTimer.current = setTimeout(() => {
-					setHighlighted(false);
-					messagesStore.highlightedMessageId = null;
-				}, 3000);
-			} else if (messagesStore.highlightedMessageId !== null) {
-				setHighlighted(false);
-			}
-		}
-
-		return () => {
-			if (highlightTimer.current) {
-				window.clearTimeout(highlightTimer.current);
-			}
-		};
-	}, [messagesStore.highlightedMessageId]);
-
-	const hoverMessage = () => {
-		hoverTimeout.current = setTimeout(() => {
-			messagesStore.setHoveredMessage(message);
-		}, 600);
-	};
-
-	const unhoverMessage = () => {
-		if (hoverTimeout.current) clearTimeout(hoverTimeout.current);
-		messagesStore.setHoveredMessage(null);
-	};
-
-	const isAttached = !!messagesStore.attachedMessages.find(
-		attMsg => attMsg.messageId === message.messageId,
-	);
-
-	function toogleMessagePin() {
-		selectedStore.toggleMessagePin(message);
-	}
-
-	return (
-		<MessageCardBase
-			message={message}
-			viewType={viewType}
-			setViewType={setViewType}
-			isHighlighted={isHighlighted}
-			hoverMessage={hoverMessage}
-			unhoverMessage={unhoverMessage}
-			isBookmarked={isBookmarked}
-			isAttached={isAttached}
-			isContentBeautified={isContentBeautified}
-			isSoftFiltered={isSoftFiltered}
-			toogleMessagePin={toogleMessagePin}
-			isDetailed={isDetailed}
-			sortOrderItems={sortOrderItems}
-		/>
-	);
-});
+// interface Props extends OwnProps, RecoveredProps {}
```

Comment:
by commenting this component you disabled highlighting and soft filtering.

we need this component, move this `message.body.map` code inside this component